shared/modules/doc/README.md

```diff
@@ -57,36 +163,61 @@ A base module is a module that implements a common interface, exposing the most
 └── types                   # Common types for the modules
 ```
 
-## Modules in detail
+## (Sub)Modules in detail
 
 We structured the code so that each module has its interface (and supporting interfaces, if any) defined in the `shared/modules` package, where the file containing the module interface follows the naming convention `[moduleName]_module.go`.
 You can start by looking at the interfaces of the modules we already implemented to get a better idea of what a module is and how it should be structured.
 You might notice that these files include `go:generate` directives, these are used to generate the module mocks for testing purposes.
 
 ### Module creation
 
-Module creation uses a typical constructor pattern signature `Create(bus modules.Bus, options ...modules.ModuleOption) (modules.Module, error)` where `options ...modules.ModuleOption` is an optional variadic argument that allows for the passing of options to the module.
-This is useful to configure the module at creation time and it's usually used during prototyping and in "sub-modules" that don't have a specific configuration file and where adding it would add unnecessary complexity and overhead. If a module has a lot of `ModuleOption`s, at that point a configuration file might be advisable.
+Module and submodule constructors both conform to the pattern established by the [factory interfaces](#factory-interfaces).
 
-To formalize the module creation convention, we've introduced a set of factory interfaces in [`shared/modules/factory.go`](https://github.com/pokt-network/pocket/tree/main/shared/modules/factory.go).
-These interfaces allow for more flexible module (or "sub-module"; see: [`RainTreeFactory`](https://github.com/pokt-network/pocket/tree/main/p2p/raintree/network.go)) creation and configuration, while maintaining a clear and concise API.
+In the case of modules, `ModuleFactoryWithConfig` MUST be applied, enforcing the following constructor signature:
```

Comment:
Is there any chance you can update the mermaid diagrams to showcase what's a MUST, SHOULD and MAY?